source/_posts/2017-11-02-secure-shell-tunnel.markdown

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "Home Assistant and SSH"
+description: "Accessing Home Assistant through a secure shell tunnel."
+date: 2017-11-02 10:00:00 +0200
+date_formatted: "November 02, 2017"
+author: Fabian Affolter
+author_twitter: fabaff
+comments: true
+categories: How-To
+og_image: /images/blog/2017-11-ssh/social.png
+---
+
+Most system engineers are very familiar with [SSH (Secure shell)](https://en.wikipedia.org/wiki/Secure_Shell). This tool which contains a server part and a client part is used to access a remote system in a secure way. It can also help you if your are running Home Assistant but don't want to expose it to the public. On a Linux system SSH is often available by default. If you are using a Windows installation additional steps are required with are not covered here.
+
+In this blog post we are going to use the tunneling option of SSH to create a secure connection and forward the Home Assistant frontend to a local system. 
+
+<!--more-->
+
+The involved parties are:
+
+- **Remote system**: Where Home Assistant is running, usually in your home network.
+- **Local system**: Where you want to see the frontend.
+
+The prerequirements are that you need to allow the forwarding of port 22 from your router to the system where Home Assistant is running in your network. It might also be needed that you enable the SSH daemon by `$ sudo systemctl start sshd` on the remote system and to adjust the host firewall. If you are running [Hass.io](https://home-assistant.io/hassio/) then enable the [SSH Server add-on](https://home-assistant.io/addons/ssh/). You must also have a public IP address or hostname which can be provided by dynamic DNS (e.g., [NO-IP](https://www.noip.com/) or [DuckDNS](https://www.duckdns.org/)).
+On your local system you need only a SSH client and you need to be in a network where SSH is allowed.
+
+First let's have a look at the command we are going to use. Use `man ssh` to get more information.
+
+```bash
+$ ssh -L 8000:localhost:8123 user@[IP_ADDRESS_REMOTE]
```

Comment:
This can also be done on Android using a SSH client such as [JuiceSSH](https://juicessh.com/) and could a good addition or maybe a good followup blog post.

Reply:
We can do a follow-up blog post for clients but I guess that you could do it on iOS or Windows Mobile too. My idea was more to show how it could be done without going into details for specific platforms.